### ISSUE TITLE
Properly show cart discounts

### ISSUE THREAD

**Issue by nikola-mladenovic** [commercetools/commercetools-sunrise-ios#31]
For line item discounts, on the cart screen, for each line item we show the price before the discount, with a strikethrough.

The UX problem arises when we have a cart discount, and it applies to some line items. Currently, we don't show the old price with a strikethrough, just a lower price in total section.
The issue here is that the cart discount sometimes won't be applied to the entire quantity of that particular line item, so how do we show that? Also, it's possible to have line item discount + cart discount for a couple of pieces of that item as well. Another tricky use case for user experience.

To sum up the current UX visually:

Cell with line item discount:
<img width="365" alt="screen shot 2016-06-15 at 12 39 13 pm" src="https://cloud.githubusercontent.com/assets/14024032/16076949/33c07eb8-32f6-11e6-97cf-13fcf67abd1b.png">

Cell with cart discount applied:
<img width="371" alt="screen shot 2016-06-15 at 12 41 48 pm" src="https://cloud.githubusercontent.com/assets/14024032/16077014/911f9148-32f6-11e6-8760-b0d0789d7945.png">

**Comment by mmoelli** [comment]
Hey @nikola-mladenovic 

From a user perspective this is a bit tricky: In the end I want to see what I will spent but also distributed among items. So my proposal would be:

Total price: always use the totalPrice of the lineItem
Price: Have some kind of logic in display:
1) If a discountedPricePerQuantity is present: use the average item price calculated based on these prices - strikethrough lineItems.price.value
2) If the price has a discounted product price: use lineitems.price.discounted.value - strikethrough lineItems.price.value
3) If there is no discount at all: use lineItems.price.value - no strikethrough

WDYT?

**Comment by lauraluiz** [comment]
@mmoelli regarding the discountedPricePerQuantity, how is it usually presented to the customer from a marketing perspective? So a shop that uses this kind of discounts, how do they advertise that they have  a product with this offer?

**Comment by mmoelli** [comment]
@lauraluiz Good question. I guess it is not, as it is a cart discount. I will rework my proposal from above.

1) If the price has a discounted product price: 
- Price: use lineitems.price.discounted.value 
- strikethrough lineItems.price.value
- Total Price: lineitems.price.discounted.value \* quantity

2) If there is no discount at all: 
- Price: use lineItems.price.value 
- no strikethrough
- Total Price: lineItems.price.value  \* quantity

discountedPricePerQuantity will be not be used, as it is more for internal calculation (distribute the discount among lineItems). Just show the discounted amount in one line at the bottom of the cart as "Order Discount"
